csrc/host_ir/ir.cpp

```diff
@@ -134,17 +134,25 @@ LaunchKernel::LaunchKernel(
     IrBuilderPasskey passkey,
     int64_t group_id,
     const LaunchParams& launch_constraints,
-    const CompileParams& compile_params,
+    CompiledKernel* compiled_kernel,
     const std::vector<Val*>& inputs,
     const std::vector<Val*>& outputs,
     Val* cache_id)
-    : Expr(passkey, inputs, outputs, {}) {
+    : Expr(passkey, inputs, outputs, {}), compiled_kernel_(compiled_kernel) {
+  NVF_CHECK(
+      compiled_kernel != nullptr,
+      "LaunchKernel requires a non-null CompiledKernel pointer");
+  NVF_CHECK(cache_id != nullptr, "LaunchKernel requires a non-null cache_id");
+
   addDataAttribute(group_id);
   addDataAttribute(launch_constraints);
-  addDataAttribute(compile_params);
+  addDataAttribute(compiled_kernel->compileParams());
```

Comment:
Do we still need this as an attribute given compiled_kernel is already a attribute? `DataAttribute`s, I believe, are attributes that are by default copied. Since you added a custom clone constructor, I don't think compileParams needs to be a `DataAttribute` any more. cc @naoyam to verify my understanding

Reply:
I commented on the header file, but I'm concerned about not having a  `compiled_kernel` itself as an attribute.